### ISSUE TITLE
min(0, t). Type inference issue with number const

### ISSUE THREAD

**Issue by vird** [AssemblyScript/assemblyscript#896]
This works fine

    export function main(t:i32):i32 {
      return min(t, 0);
    }

But if const is first

    export function main(t:i32):i32 {
      return min(0, t);
    }

It can't handle properly, and 0 is used as f64 const

    ERROR AS200: Conversion from type 'f64' to 'i32' requires an explicit cast.

       return min(0, t);
              ~~~~~~~~~
     in index.ts(2,9)

    ERROR: Compile error
        at Object.main (/data/nodejs/experiments/as_hotreload_env/node_modules/assemblyscript/cli/asc.js:530:21)
        at Object.<anonymous> (/data/nodejs/experiments/as_hotreload_env/node_modules/assemblyscript/bin/asc:21:26)
        at Module._compile (internal/modules/cjs/loader.js:816:30)
        at Object.Module._extensions..js (internal/modules/cjs/loader.js:827:10)
        at Module.load (internal/modules/cjs/loader.js:685:32)
        at Function.Module._load (internal/modules/cjs/loader.js:620:12)
        at Function.Module.runMain (internal/modules/cjs/loader.js:877:12)
        at internal/main/run_main_module.js:21:11

**Comment by MaxGraey** [comment]
Can confirm

**Comment by MaxGraey** [comment]
Should be fixed in #904
